src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Cohort/DetailsViewModelMapper.cs

```diff
@@ -265,5 +265,82 @@ private string GetFundingBandExcessLabel(int apprenticeshipsOverCap)
                 return new string("The price for these apprenticeships is above the");
             return null;
         }
+
+        private string GetCohortStatus(GetCohortResponse cohort, IReadOnlyCollection<DraftApprenticeshipDto> draftApprenticeships)
+        {
+            if (cohort.TransferSenderId.HasValue &&
+                cohort.TransferApprovalStatus == CommitmentsV2.Types.TransferApprovalStatus.Pending)
+            {
+                if (cohort.WithParty == CommitmentsV2.Types.Party.TransferSender)
+                {
+                    return "Pending - with funding employer";
+                }
+                else if (cohort.WithParty == CommitmentsV2.Types.Party.Employer)
+                {
+                    return GetEmployerOnlyStatus(cohort);
+                }
+                else if (cohort.WithParty == CommitmentsV2.Types.Party.Provider)
+                {
+                    return GetProviderOnlyStatus(cohort);
+                }
+            }
+            else if (cohort.IsApprovedByEmployer && cohort.IsApprovedByProvider)
+            {
+                return "Approved";
+            }
+            else if (cohort.WithParty == CommitmentsV2.Types.Party.Provider)
+            {
+                return GetProviderOnlyStatus(cohort);
+            }
+            else if (cohort.WithParty == CommitmentsV2.Types.Party.Employer)
+            {
+                return GetEmployerOnlyStatus(cohort);
+            }
+
+            return "New request";
+        }
+
+        private static string GetEmployerOnlyStatus(GetCohortResponse cohort)
+        {
+            if (cohort.LastAction == CommitmentsV2.Types.LastAction.None)
+            {
+                return "New request";
+            }
+            else if (cohort.LastAction == CommitmentsV2.Types.LastAction.Amend)
+            {
+                return "Ready for review";
+            }
+            else if (cohort.LastAction == CommitmentsV2.Types.LastAction.Approve)
+            {
+                if (!cohort.IsApprovedByProvider && !cohort.IsApprovedByEmployer)
+                    return "Ready for review";
+
+                return "Ready for approval";
+            }
+            else
```

Comment:
what if cohort.LastAction == LastAction.AmendAfterRejected ?

This would currently fall into the catch all else{} at the bottom. There is no test for this scenario

Reply:
done